pandas/_libs/tslibs/offsets.pyx

```diff
@@ -1157,7 +1157,9 @@ cdef class RelativeDeltaOffset(BaseOffset):
             return dt64other
         elif not self._use_relativedelta and hasattr(self, "_offset"):
             # timedelta
-            delta = Timedelta(self._offset * self.n)
+            num_nano = getattr(self, "nanoseconds", 0)
+            rem_nano = Timedelta(nanoseconds=num_nano)
```

Comment:
I think for performance reasons, it would be good to only construct the Timedelta and add to the offset if `nanoseconds != 0`